bob-apps/bin/bobd.rs

```diff
@@ -144,6 +148,51 @@ fn spawn_signal_handler(
     Ok(())
 }
 
+fn check_folders(node: &NodeConfig, init_flag: bool) {
+    if let BackendType::Pearl = node.backend_type() {
+        let root_dir = node.pearl().settings().root_dir_name();
+        let alien_dir = node.pearl().settings().alien_root_dir_name();
+
+        let p_mutex = node.disks();
+        let paths = p_mutex.lock().expect("node disks mutex");
+        for i in 0..paths.len() {
+            let mut bob_path = PathBuf::from(paths[i].path());
+            bob_path.push(root_dir);
+            let bob_path = bob_path.as_path();
+            if !bob_path.is_dir() {
+                if init_flag {
+                    create_dir(bob_path).expect("Failed to create bob folder");
+                } else {
+                    if let Some(path_str) = bob_path.to_str() {
+                        error!("{} folder doesn't exist, try to use --init_folders flag", path_str);
+                        panic!("{} folder doesn't exist, try to use --init_folders flag", path_str);
+                    } else {
+                        error!("bob folder doesn't exist, try to use --init_folders flag");
```

Comment:
Under what circumstances can we enter the `else` path?

Reply:
We enter 'else' branch if path to root dir can't be turned into string

Reply:
So the next obvious question: when can the path to root dir not be converted to string? This seems to be an impossible situation

Reply:
Seems like it. So I'll remove 'if'